x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -146,6 +181,172 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                 return { createdAlerts: [], errors: {}, alertsWereTruncated: false };
               }
             },
+            alertWithSuppression: async (
+              alerts,
+              suppressionWindow,
+              enrichAlerts,
+              currentTimeOverride
+            ) => {
+              const ruleDataClientWriter = await ruleDataClient.getWriter({
+                namespace: options.spaceId,
+              });
+
+              // Only write alerts if:
+              // - writing is enabled
+              //   AND
+              //   - rule execution has not been cancelled due to timeout
+              //     OR
+              //   - if execution has been cancelled due to timeout, if feature flags are configured to write alerts anyway
+              const writeAlerts =
+                ruleDataClient.isWriteEnabled() && options.services.shouldWriteAlerts();
+
+              if (writeAlerts && alerts.length > 0) {
+                const suppressionWindowStart = dateMath.parse(suppressionWindow, {
+                  forceNow: currentTimeOverride,
+                });
+                if (!suppressionWindowStart) {
+                  throw new Error('Failed to parse suppression window');
+                }
+
+                const suppressionAlertSearchRequest = {
```

Comment:
Can we make this query work with `ALERT_START` instead of `TIMESTAMP`? It will make it easier for the framework to enable suppression for O11y and stack alerts too (given timestamp behaves differently in O11y / stack vs security solution) we need to look at the first detected time.

We will need to set `ALERT_START` for newly detected alerts within `augmentAlerts` to make this work.